### DISCUSSION TITLE
SSKR threshold query

### DISCUSSION THREAD

**Discussion by aido** [BlockchainCommons/Gordian-Developer-Community#129] [Q&A] [answered]
Hi,

The following lines of code in the [bc-sskr](https://github.com/BlockchainCommons/bc-sskr) repo would suggest that a share count greater than 1 should not have a threshold of 1 i.e. 1-of-1 IS allowed but 1-of-2 is NOT allowed.

https://github.com/BlockchainCommons/bc-sskr/blob/84e1e76fda3155fd8805c1618cb56fb42f9681c7/src/encoding.c#L131-L132

Although I understand why such a condition would exist a quick scan of the SSKR documentation shows no mention of it.

So is this correct;  1-of-1 IS allowed but 1-of-2 is NOT allowed.? Have I missed this condition in the documentation?

**Comment by wolfmcnally** [comment, marked as answer]
@aido 
In a 1-of-N split, every share is equivalent to the secret. This codebase allows a 1-of-1 split but disallows 1-of-N in general because it's not secure. Our Rust SSKR codebase does allow 1-of-N splits, but we don't recommend using them for the same reason: every share is the whole secret, which defeats the purpose of SSKR.

@ChristopherA

**Reply by aido** [reply]
@wolfmcnally thanks for the response.

So a 1-of-n split where n > 1 is not forbidden by the standard but is discouraged. I shall have my [app-seed-tool](https://github.com/BlockchainCommons/Gordian-Developer-Community/discussions/app-seed-tool) Ledger application not allow it.

By the way, off topic but I am eagerly awaiting the latest version of [seedtool-cli-rust](https://github.com/BlockchainCommons/seedtool-cli-rust). Nice work so far. I am particularly interested in it supporting version 2 sskr with the new CBOR tag.
